R-package/R/lgb.Booster.R

```diff
@@ -49,7 +49,11 @@ Booster <- R6Class(
       }
       class(handle)     <- "lgb.Booster.handle"
       private$handle    <- handle
-      private$num_class <- 1L
+      if ("num_class" %in% names(params)){
+	private$num_class <- params$num_class
+      } else{
+	private$num_class <- 1L
+      }
       private$num_class <-
         lgb.call("LGBM_BoosterGetNumClasses_R", ret = private$num_class, private$handle)
```

Comment:
Workflow is like this:

* Get param list
* Apply param list to booster
* (...)
* num_class is set set to default value (if it doesn't exist, then it will be 1 at the end because it is the default value)
* num_class is harvested from booster else it gives the default value

It should be working already that way, can you post a reproducible example?